### ISSUE TITLE
Glue Jobs fail with ERROR when AvailabilityZone is specified

### ISSUE THREAD

**Issue by evilensky** [aws-cloudformation/cfn-lint#3886]
### CloudFormation Lint Version

1.12.3

### What operating system are you using?

Mac and Linux

### Describe the bug

GlueConnections fail linting when az is specified:

```
INFO     cfnlint.runner:runner.py:125 Run scan of template tests/data/integration/based-on-poc-serverless-service.yml
ERROR    lint_cfn:lint_cfn.py:17 W3010: Availability zone properties should not be hardcoded: Avoid hardcoding availability zones 'us-west-2c'
 at:Resources/AlsoPoliticianProcessingDefaultGlueConnection/Properties/ConnectionInput/PhysicalConnectionRequirements/AvailabilityZone
```

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-connection-physicalconnectionrequirements.html#cfn-glue-connection-physicalconnectionrequirements-availabilityzone

What's not documented is that AvailabilityZone is required if SecurityGroupIdList[] is provided or the Glue Connection silently fails.

### Expected behavior

Should not lint as error.

### Reproduction template

```yaml
Resources:
  AlsoPoliticianProcessingDefaultGlueConnection:
    Properties:
      CatalogId:
        Ref: AWS::AccountId
      ConnectionInput:
        ConnectionType: NETWORK
        Description: Default VPC Connection for job also-politician-processing
        Name: tools-dev-poc-serverless-service-also-politician-processing-network
        PhysicalConnectionRequirements:
          AvailabilityZone: us-west-2c
          SecurityGroupIdList:
          - Fn::ImportValue: DefaultGlueSecurityGroupId
          SubnetId:
            Fn::ImportValue: DefaultGlueSubnetId
```

**Comment by kddejong** [comment]
This is a warning level rule pointing toward template re-usability.

You can solve these type of things by doing:
```yaml
Parameters:
  AvailabilityZone:
    Type: String
    Default: us-west-2c

Resources:
  AlsoPoliticianProcessingDefaultGlueConnection:
    Type: AWS::Glue::Connection
    Properties:
      CatalogId:
        Ref: AWS::AccountId
      ConnectionInput:
        ConnectionType: NETWORK
        Description: Default VPC Connection for job also-politician-processing
        Name: tools-dev-poc-serverless-service-also-politician-processing-network
        PhysicalConnectionRequirements:
          AvailabilityZone: !Ref AvailabilityZone
          SecurityGroupIdList:
          - Fn::ImportValue: DefaultGlueSecurityGroupId
          SubnetId:
            Fn::ImportValue: DefaultGlueSubnetId
```

or 
```yaml
Resources:
  AlsoPoliticianProcessingDefaultGlueConnection:
    Type: AWS::Glue::Connection
    Properties:
      CatalogId:
        Ref: AWS::AccountId
      ConnectionInput:
        ConnectionType: NETWORK
        Description: Default VPC Connection for job also-politician-processing
        Name: tools-dev-poc-serverless-service-also-politician-processing-network
        PhysicalConnectionRequirements:
          AvailabilityZone: !Select [3, !GetAZs ""]
          SecurityGroupIdList:
          - Fn::ImportValue: DefaultGlueSecurityGroupId
          SubnetId:
            Fn::ImportValue: DefaultGlueSubnetId
```

You can also disable the rule doing:
```yaml

Metadata:
  cfn-lint:
    config:
      ignore_checks:
      - W3010
Resources:
  AlsoPoliticianProcessingDefaultGlueConnection:
    Type: AWS::Glue::Connection
    Properties:
      CatalogId:
        Ref: AWS::AccountId
      ConnectionInput:
        ConnectionType: NETWORK
        Description: Default VPC Connection for job also-politician-processing
        Name: tools-dev-poc-serverless-service-also-politician-processing-network
        PhysicalConnectionRequirements:
          AvailabilityZone: us-west-2c
          SecurityGroupIdList:
          - Fn::ImportValue: DefaultGlueSecurityGroupId
          SubnetId:
            Fn::ImportValue: DefaultGlueSubnetId

```
